### PR TITLE
Fb asim 4157 separator efficiency

### PR DESCRIPTION
**Pull request by edersongrein**

### CODE REVIEW

File: src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py [hunk 1]
```diff
@@ -954,6 +954,8 @@ def load_separator_node_properties_description(
         "initial_phase_volume_fractions": get_dict_with_scalar_loader(
             category="volume fraction"
         ),
+        "gas_separation_efficiency": get_scalar_loader(from_unit="%"),
```
Comment (ggrbill) [inline src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py:957]: I believe you have to make something similar in the `case_to_alfacase.py` file.
@williamjamir could you confirm?

Comment (ggrbill) [reply src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py:957]: Also, would be nice to add tests for the new entries.

**Review by ggrbill — CHANGES_REQUESTED** [review]

File: src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py (hunk 1 above)
Comment (ggrbill) [reply src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py:957]: @williamjamir, I do not remember, do we need to add tests? Or the test suit makes it automatically for any new entry?

File: src/alfasim_sdk/_internal/alfacase/case_description.py [hunk 2]
```diff
@@ -1097,6 +1097,8 @@ class SeparatorNodePropertiesDescription:
             constants.FLUID_OIL: Scalar("volume fraction", 0.5, "-"),
         }
     )
+    gas_separation_efficiency = attrib_scalar(default=Scalar(1.0, "-"))
+    liquid_separation_efficiency = attrib_scalar(default=Scalar(1.0, "-"))
```
Comment (williamjamir) [inline src/alfasim_sdk/_internal/alfacase/case_description.py:1101]: I particularly think it's better to put the category explicitly and not rely on the default value (if I'm not mistaken is dimensionless)

File: src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py (hunk 1 above)
Comment (prusse-martin) [reply src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py:957]: No. `case_to_alfacase` is almost automatic.
I hope in the future to mate a bit less reliant o manual updates =)

Comment (williamjamir) [reply src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py:957]: > I believe you have to make something similar in the case_to_alfacase.py file.

Case to ALFAcase is more "smart", is just a loop over all attributes, and automatically gets the value.
You will need to update that only if a new "type" is added.

> Also, would be nice to add tests for the new entries.
Definitely, please update the `SeparatorNodePropertiesDescription` on `alfasim-sdk/tests/common_testing/alfasim_sdk_common_testing/filled_case_descriptions.py` and add these attributes there as well.

> @williamjamir, I do not remember, do we need to add tests? Or the test suit makes it automatically for any new entry?

There is a basic "round-trip" test that takes a filled case description and converts it to ALFAcase and reimport to check if the values match.
With this test, you can validate that the conversion (ALFAcase->case; case->ALFAcase) is working.

Comment (ggrbill) [reply src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py:957]: Thanks a lot, Guys!

@edersongrein if you don't understand some of the instructions, please let me know, I can help you on that.

Comment (williamjamir) [reply src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py:957]: We didn't add (yet) a smart way to automatically check if the filled case description is outdated (that would be nice).

I think that one possible solution is to add a "template" when a new PR is opened with TODOS to be check and one of the options is "Did you update the filled case description available at X ? ".

Comment (ggrbill) [reply src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py:957]: Sounds good to me. I think it would help.

**Review by williamjamir — CHANGES_REQUESTED** [review]
Please update the `SeparatorNodePropertiesDescriptio` class on
`alfasim-sdk/tests/common_testing/alfasim_sdk_common_testing/filled_case_descriptions.py`

Also, you need to run the tests with `force-regen` to update the expected values (this is the reason that the CI is failing).

File: src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py (hunk 1 above)
Comment (edersongrein) [reply src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py:957]: > Thanks a lot, Guys!
> 
> @edersongrein if you don't understand some of the instructions, please let me know, I can help you on that.

Thanks. Probably I will

File: src/alfasim_sdk/_internal/alfacase/case_description.py (hunk 2 above)
Comment (edersongrein) [reply src/alfasim_sdk/_internal/alfacase/case_description.py]: Done

**Review by ggrbill — APPROVED** [review]

**Review by williamjamir — APPROVED** [review]
Awesome! 🚀 

Just don't forget to update the HISTORY.rst with these new additions ;)

File: HISTORY.rst [hunk 3]
```diff
@@ -5,7 +5,7 @@ History
 0.9.0 (unreleased)
 ==================
 
-*
+* Add gas and liquid separation efficiency to ``Separator`` model
```
Comment (prusse-martin) [inline HISTORY.rst:8]:
```suggestion
* Add gas and liquid separation efficiency to ``Separator`` model.
```

**Review by prusse-martin — APPROVED** [review]

**Pull request merged**